content/docs/04-clusters/03-edge/30-cloud-init.md

```diff
@@ -0,0 +1,287 @@
+| `before-upgrade`    | This stage is executed before upgrade                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |      |
+| `before-reset`      | This stage is executed before reset                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |      |
+
+
+<InfoBox>
+
+ Each stage has a before and after hook you can use to achieve more granular customization. For example, you can use `network.after` to verify network connectivity.
+
+ </InfoBox>
+
+
+# Where to apply Cloud-Init Stages?
+
+You may ask yourself where to use cloud-init stages, as both the Stylus installer and the OS pack support the usage of cloud-init stages.
+
+If you want to apply specific configurations to a specific site, then use the user-data option of Stylus to provide site settings to that specific site. If you have common configurations across a fleet of edge devices, customize the OS pack and use the cloud-init stages to apply those configurations. 
```

Comment:
Maybe have those 2 as bullet points to really delineate the usage?

Reply:
Great feedback, and I agree. It looks cleaner in bullet form.